generator/lookml.py

```diff
@@ -41,196 +37,140 @@
-                    raise KeyError(dimension.name)
-                dimensions[dimension.name] = dimension
-        return list(dimensions.values())
-
-
-@dataclass
-class Measure:
-    """LookML View Measure."""
-
-    name: str
-    type: str
-    sql: Optional[str] = None
-
-    def to_lookml(self) -> str:
-        """Serialize measure as lookml."""
-        return (
-            f"measure: {self.name} {{"
-            + f"\n  type: {self.type}"
-            + (f"\n  sql: {self.sql} ;;" if self.sql is not None else "")
-            + "\n}"
-        )
-
-    @classmethod
-    def from_dimensions(klass, dimensions: List[Dimension]) -> List["Measure"]:
-        """Generate measures from a list of dimensions.
-
-        When no dimension-specific measures are found, return a single "count" measure.
-
-        Raise KeyError if dimensions result in duplicate measures.
-        """
-        measures = {}
-        for dimension in dimensions:
-            if dimension.name in {"client_id", "client_info__client_id"}:
-                measure = klass("clients", "count_distinct", f"${{{dimension.name}}}")
-            elif dimension.name == "document_id":
-                measure = klass("ping_count", "count")
-            else:
-                continue
-            if measure.name in measures:
-                raise KeyError(measure.name)
-            measures[measure.name] = measure
-        # return a generic count measure if no default measures were generated
-        return list(measures.values()) or [klass("count", "count")]
+        measures[name] = measure
+    # return a generic count measure if no default measures were generated
+    return list(measures.values()) or [{"name": "count", "type": "count"}]
 
 
 def _generate_views(
     client, out_dir: Path, views: Dict[str, List[Dict[str, str]]]
 ) -> Iterable[Path]:
-    for view, tables in views.items():
+    for name, tables in views.items():
+        view: Dict[str, Any] = {"name": name}
         # use schema for the table where channel=="release" or the first one
         table = next(
             (table for table in tables if table.get("channel") == "release"),
             tables[0],
         )["table"]
-        try:
-            dimensions = Dimension.from_schema(client.get_table(table).schema)
-        except KeyError as e:
-            raise click.ClickException(f"duplicate dimension {e} for table {table!r}")
-        try:
-            measures = Measure.from_dimensions(dimensions)
-        except KeyError as e:
-            raise click.ClickException(f"duplicate measure {e} for table {table!r}")
-        fields = []
-        sql_table_name = table
+        # add dimensions and dimension groups
+        dimensions = _generate_dimensions(client, table)
+        view["dimensions"] = [d for d in dimensions if "timeframes" not in d]
+        view["dimension_groups"] = [d for d in dimensions if "timeframes" in d]
```

Comment:
I think we need a more robust method of determining if a dimension is a dimension_group ([docs](https://docs.looker.com/reference/field-params/dimension_group#intervals)). Reading that, it should be that either `timeframes` is in d or `intervals` is in d. Maybe we should pull this to a separate function, e.g.
```
view["dimensions"] = filter(d, lambda x: not is_dimension_group(x))
view["dimension_groups"] = filter(d, is_dimension_group)
```

Reply:
done, with one difference: used `filterfalse` from the standard library's `itertools`